NBXplorer/Repository.cs

```diff
@@ -648,6 +652,40 @@ public async Task SaveKeyInformations(KeyPathInformation[] keyPathInformations)
 			await tx.Commit();
 		}
 
+		public async Task SaveOutPointToScript((OutPoint outPoint, Script script)[] mapping)
+		{
+			using var tx = await engine.OpenTransaction();
+			foreach (var (outPoint, script) in mapping)
+			{
+				var bytes = ToBytes(script);
+				await GetOutPointsIndex(tx, outPoint).Insert(0, bytes);
+			}
+			await tx.Commit();
+		}
+		public async Task<Dictionary<OutPoint, Script>> GetOutPointToScript(OutPoint[] outPoints)
+		{
+			var result = new Dictionary<OutPoint, Script>();
+			if (outPoints.Length == 0)
+				return result;
+			foreach (var batch in outPoints.Batch(BatchSize))
+			{
+				using var tx = await engine.OpenTransaction();
+				foreach (var outPoint in batch)
+				{
+					var table = GetOutPointsIndex(tx, outPoint);
+					var rawScript = await table.SelectBytes(0);
+					if (rawScript is null)
+						result.Add(outPoint, null);
```

Comment:
Given that the majority will end up in this case, you should not do this.
Imagine a block has around 3000 outpoints. This would add 3000 elements to the dictionary. (and with unset capacity on initialization, it will internally create lot's of copy of arrays, which will probably slow down things)

Reply:
Fixed this